lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -217,13 +218,19 @@ IncrementalJIT::getSymbolAddressWithoutMangling(llvm::StringRef Name,
   if (auto Sym = getInjectedSymbols(Name))
     return Sym;
 
+  const std::string SymName = Name;
```

Comment:
And the problem was dlopen needs a c-string, and if you use StringRef you're basically constructing an llvm::StringRef which is cheap enough, but then you have to re-construct a new std::string to get the c-string.